keras/optimizers.py

```diff
@@ -410,8 +410,8 @@ def get_updates(self, loss, params):
             self.updates.append(K.update(a, new_a))
 
             # use the new accumulator and the *old* delta_accumulator
-            update = g * K.sqrt(d_a + self.epsilon) / K.sqrt(new_a + self.epsilon)
-            new_p = p - lr * update
+            update = lr * g * K.sqrt(d_a + self.epsilon) / K.sqrt(new_a + self.epsilon)
```

Comment:
How is this not exactly equivalent to the previous code?

Reply:
Look at the line 423, you are accumulating the root mean squared value of the update; if `lr=1` the code is exactly equivalent, but if `lr!=1` the accumulated value in the old version does not take `lr` into account

Reply:
Look at the line 423, you are accumulating the root mean squared value of the update; if `lr=1` the code is exactly equivalent, but if `lr!=1` the accumulated value in the old version does not take `lr` into account

Reply:
Am I clear? :-D